engine/src/test/java/org/hibernate/validator/test/cfg/ConstraintMappingTest.java

```diff
@@ -145,7 +149,7 @@ public void testSingleConstraint() {
 
 		Set<ConstraintViolation<Marathon>> violations = validator.validate( new Marathon() );
 		assertNumberOfViolations( violations, 1 );
-		assertConstraintViolation( violations.iterator().next(), "may not be null" );
+		assertConstraintViolation( violations.iterator().next(), NotNull.class, "may not be null" );
```

Comment:
> we test the constraint type and the path independently. We could have a mix-up between the constraint type and the path and it would not be reported as an error.

Indeed, that's nagging me, too. I've filed HV-1350, please feel free to generalize this one for related improvements around assertions.